src/error.rs

```diff
@@ -129,6 +129,8 @@ pub enum RoutingError {
     ExceedsRateLimit(Digest256),
     /// Invalid configuration
     ConfigError(ConfigFileHandlerError),
+    /// Failed to initialize crypto library.
+    CryptoInitError(safe_crypto::Error),
```

Comment:
I think it'd be more canonical to just have a single `RoutingError` variant which holds a `safe_crypto::Error` (e.g. `RoutingError::Crypto(safe_crypto::Error)` and then implement `From<safe_crypto::Error> for RoutingError`.

That lets us just use `?` rather than having to use `map_err()`.